src/hashstore/hashstore.py

```diff
@@ -70,6 +70,19 @@ def _set_sysmeta(self, pid, sysmeta, obj_cid):
             file.write(sysmeta)
         return s_cid
 
+    def _get_sysmeta(self, pid):
+        """Returns sysmeta content given persistent identifier (pid)"""
+        s_cid = self.hash_string(pid)
+        s_path = Path(self.abs_path(s_cid))
+        s_content = s_path.read_text()
+        cid = s_content[:64]
+        return cid
+
+    def retrieve(self, pid):
```

Comment:
The return value for retrieve should be the content of the cid, not the cid itself.